packages/solidity/contracts/StakingRewardsStore.sol

```diff
@@ -36,6 +36,8 @@ contract StakingRewardsStore is IStakingRewardsStore, AccessControl, Utils, Time
 
     uint32 private constant PPM_RESOLUTION = 1000000;
 
+    uint256 private constant MAX_REWARD_RATE = 2**128 - 1;
+
```

Comment:
So how exactly do we leverage from that?
I'd expect to see an actual state variable in this contract changed from `uint256` to `uint128`.

Reply:
This change's goal wasn't gas optimizations but rather to prevent an accident where the rewards rate is set to a too high value. The `2**128 - 1` is a somewhat arbitrary safe value (which can even be lower, btw). Reducing only this variable to 128 bits will involve too many unrelated changes, so we should only consider reducing it (and other members) during a gas optimization and store migration overhaul.